libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/AlternativeRouteInfo.kt

```diff
@@ -8,6 +8,8 @@ import com.mapbox.navigation.core.directions.session.RoutesObserver
  * @param distance distance (based on the referenced route)
  * @param duration duration (based on the referenced route)
  */
+// the values might get out of date after route refresh,
```

Comment:
Should we add a note to the KDocs? 🤔

Reply:
We still have a point from https://github.com/mapbox/mapbox-navigation-android/pull/5653 to address:
> The new AlternativeRouteMetadata goes out-of-date whenever RoutesObserver or NavigationRouteAlternativesObserver fires and is documented as such but I'm wondering whether we should bake in a isValid flag into the metadata instances to better surface this state. Additionally, this open up a question whether NavigationRouteAlternativesObserver should invalidate the metadata at all, it might just be need to invalidate metadata when MapboxNavigation#setRoutes is called and the primary route changes.

Let's do that together.